pyfr/integrators/dual/phys/controllers.py

```diff
@@ -47,8 +49,21 @@ def advance_to(self, t):
             raise ValueError('Advance time is in the past')
 
         while self.tcurr < t:
+
+            if self.tcurr + self._dt < t < self.tcurr + 2*self._dt:
```

Comment:
I don't really feel like this is the best strategy. If target t is something like n*dt+0.01dt, I think it will take a while to go down to that level. And it will be even worse if t is n*dt+0.001dt and so on. You actually provided a plot for the previous strategy (tcurr+2dt<t<tcurr+3dt), but not for the current one. Can you do a plot where t%dt varies from 0.01dt to 0.99dt and show how the dt changes and how many steps it takes until getting to the target t?

What I had in mind was more like limiting the variation in t with a min and max multiplier such as 0.9 and 1.1. And as you march in time and get close to target t, say when you're ~10dt away from it, you just fix dt by multiplying a coefficient between 0.9 and 1, so that target t can be achieved without changing dt again.

Reply:
Instead of changing dt within `[dt,2*dt]` prior to desired time, I am now changing within `[dt,i*dt]` prior to the desired time. `i` here is set by the config file. 
In total `dt` is updated thrice, once when it is near the desired time, next when it needs to exactly hit the desired time in a single step, and finally when it is equal to or greater than the desired time.